### ISSUE TITLE
_p_resolveConflict() should be called even if a conflicting change doesn't change the state

### ISSUE THREAD

**Issue by jmuchemb** [zopefoundation/ZODB#97]
What I'd like is to commit the following change:
```diff
--- a/src/ZODB/ConflictResolution.py
+++ b/src/ZODB/ConflictResolution.py
@@ -271,13 +271,6 @@ def tryToResolveConflict(self, oid, committedSerial, oldSerial, newpickle,
         if not committedData:
             committedData  = self.loadSerial(oid, committedSerial)
 
-        if newpickle == oldData:
-            # old -> new diff is empty, so merge is trivial
-            return committedData
-        if committedData == oldData:
-            # old -> committed diff is empty, so merge is trivial
-            return newpickle
-
         newstate = unpickler.load()
         old       = state(self, oid, oldSerial, prfactory, oldData)
         committed = state(self, oid, committedSerial, prfactory, committedData)
```
(i.e. revert to the behaviour of ZODB 3.10.3-)

For example, we need such behaviour to implement an "asynchronous" cache:
- initially, a cache value is not filled (e.g. `None` is used to describe this state)
- a transaction fills the cache (actually done by a background zope) (`None -> "foo"`)
- a concurrent transaction invalidates the cache due to some user action (`None -> None`), and pushes a new background task to fill the cache

In our case, the final value must be `None` and we implement `_p_resolveConflict()` for that. However, with ZODB 3.10.4+, `None -> None` is seen as "no change" and the result is `"foo"`.

**Comment by jimfulton** [comment]
+1

**Comment by jmuchemb** [comment]
Fixed by #98
